package.json

```diff
@@ -9,7 +9,7 @@
   "private": true,
   "scripts": {
     "build": "yarn build:client",
-    "build:client": "yarn clean && webpack --env development",
+    "build:client": "yarn clean && webpack --env production",
```

Comment:
I had to set this to `development` and make alterations to the Dockerfile (https://github.com/project-koku/koku-ui/blob/master/Dockerfile) when we added the PF3 ListView components. I would confirm that the cost details page still works and you may look at updating the Dockerfile as it would be good to get back to building as production there as that will make the app much smaller.

Reply:
Yes, the `Cost Details` page doesn't render and I see this error message in the console:

```
TypeError: Cannot read property 'displayName' of undefined
    at Toolbar.js:18
    at Array.filter (<anonymous>)
    at h (Toolbar.js:17)
    at Fa (react-dom.production.min.js:173)
    at pi (react-dom.production.min.js:202)
    at hi (react-dom.production.min.js:203)
    at Yi (react-dom.production.min.js:212)
    at Ki (react-dom.production.min.js:211)
    at $i (react-dom.production.min.js:211)
    at Vi (react-dom.production.min.js:209)
```

Reply:
Update: when using `production` in webpack `patternfly-react` stops working:
 - components like `Filter.TypeSelector` are undefined
 - `Toolbar` fails to render

To fix the first problem, call to `FilterTypeSelector`, `ListViewItem` and etc instead of `Filter.TypeSelector`, ListView.Item` and etc respectively.

I still didn't troubleshoot the second problem as I have no way to investigate it.
Trying to change the code in `node_module` seems to make any change,

I'm not sure whether it's related to our `webpack.config.js` or a problem on `Patternfly` side.

Reply:
@dlabrecq  You think its worth reaching out to @dmiller9911 on this ^ ?

He may have some thoughts since he did the webpack setup.

Reply:
My guess for this would be a PatternFly side with how it exports those subcomponents by mutating the module in the index.js. there is a `"sideEffects": false` false param in package.json would you be able to try and change that to `true` in `node_modules/patternfly-react/package.json` and retry. It is possible that could break the tree shaking. Webpack may think that it is not used and remove it. If that works patternfly-react will need that set to true which will sadly remove all of tree shaking for it causing the whole package to be imported even if it is not all used. The other option for it would be to stop doing that. Rollup did not support it at all so it would be a good change, just a big breaking one. Most consumers of `patternfly-react` at using `< v4 webpack` which does not support the sideEffects flag. So it is likely that many do not see the issue.

Reply:
I tried setting `"sideEffects": true` with PF3, PF-NEXT. For kicks, I also tried react-core, react-icons, react-styles, & react-tokens with no effect.

When I turn off hot deploy, I can at least see the dashboard and the following errors on the cost details page.

![screen shot 2018-09-26 at 6 55 42 pm](https://user-images.githubusercontent.com/17481322/46113872-17ad9900-c1be-11e8-85be-b07bcbbf0ca2.png)

Reply:
I haven't worked on it much today but I think I have a way to troubleshoot it.
I will try working on it tomorrow. Nevertheless, if it takes too long, it's a good idea to postpone it till we move `detailsToolbar` to PF-next.

WDYT?

Reply:
OK - I found the source of the problem.
As mentioned by @dmiller9911 it's related to the subcomponents.

The Toolbar is using `Grid.Row` and `Grid.Col` which are in production `undefined` causing the entire page to fail and seeing those two error messages.

Reply:
BTW - changing the `sideEffect` flag to `true` fixed that problem.